lib/HLSL/DxilPreparePasses.cpp

```diff
@@ -657,16 +657,23 @@ class DxilFinalizeModule : public ModulePass {
           // In spite of the <=1 assert above, loop here in case the assumption is wrong
           for (auto II = U->user_begin(), EE = U->user_end(); II != EE;) {
             User *UU = *II++;
-            BranchInst *BI = cast<BranchInst>(UU);
-            Function *F = BI->getParent()->getParent();
+            Instruction *I = dyn_cast<Instruction>(UU);
+            if (!I) continue;
+            Function *F = I->getParent()->getParent();
```

Comment:
Function can be retrieved from the CallInst *CI outside loop, right?  That means we don't need `I` here.